spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -337,14 +337,21 @@ private Result doExecuteWithPartialProgress(
         commitManager.service(groupsPerCommit);
     commitService.start();
 
+    List<FileGroupFailureResult> rewriteFailures = Lists.newArrayList();
```

Comment:
Race Condition here, need to use a concurrent structure

Reply:
Thanks RussellSpitzer for pointing out this problem.